### PR TITLE
Query server for entity (#18)

### LINKED ISSUE #18: Query server for `entity`
As a User
I want to be able to query XTDB for `entity`
So I need support for entity query https://docs.xtdb.com/clients/1.23.2/clojure/#_entity
https://docs.xtdb.com/clients/http/#entity

Acceptance Criteria:

* Server supports entity query

### PR DESCRIPTION
**Pull request by pfeodrippe**
Fixes #18.

### CODE REVIEW

**Comment by pfeodrippe** [comment]
@naomijub Let me know if this is what you were talking about when you said that you just wanted to return a string for now o/

File: resources/entity.proto [hunk 1]
```diff
@@ -18,4 +18,19 @@ message EntityTxRequest {
...
-}
\ No newline at end of file
+}
+
+message EntityResponse {
+    string xt_id = 1;
+    string content = 2;
+}
+
+message EntityRequest {
```
Comment (naomijub) [inline resources/entity.proto:28]: Maybe we could just use the same type for this and `EntityTxRequest`. Would stick with `EntityRequest`

File: src/gxtdb/adapters/entity.clj [hunk 2]
```diff
@@ -2,8 +2,12 @@
...
\ No newline at end of file
+  {:xt-id (-> entity-tx-response :xt/id str)
+   :content-hash (-> entity-tx-response :xtdb.api/content-hash str)
+   :valid-time (-> entity-tx-response :xtdb.api/valid-time utils/->inst-str)
+   :tx-time (-> entity-tx-response :xtdb.api/tx-time utils/->inst-str)
+   :tx-id (:xtdb.api/tx-id entity-tx-response)})
+
+(defn entity->proto [entity-response]
+  {:xt-id (-> entity-response :xt/id str)
+   :content (pr-str entity-response)})
```
Comment (naomijub) [inline src/gxtdb/adapters/entity.clj:13]: here you should extract the `:content` key from the `entity-response` before parsing the edn

File: test/gxtdb/service_test.clj [hunk 3]
```diff
@@ -11,11 +11,11 @@
 
 ;; Setup.
 (def ^:dynamic *opts* {})
-(def node (xt/start-node *opts*))
+(defonce node (xt/start-node *opts*))
```
Comment (naomijub) [inline test/gxtdb/service_test.clj:14]: I tried this change in the past and it failed some tests, but well, now it all seems to be working

**Review by naomijub — COMMENTED** [review]
Looks good, just a couple of comments to review

File: src/gxtdb/adapters/entity.clj (hunk 2 above)
Comment (pfeodrippe) [reply src/gxtdb/adapters/entity.clj:13]: There is no :contents key officially, the clj return is just a map, this what I’m doing is just so we can have the id available (which will also exist inside contents, but in it’s raw form), does it make sense?

File: test/gxtdb/service_test.clj (hunk 3 above)
Comment (pfeodrippe) [reply test/gxtdb/service_test.clj:14]: Nice!

File: resources/entity.proto (hunk 1 above)
Comment (pfeodrippe) [reply resources/entity.proto:28]: Yeah, I didn’t want to use the same as they may diverge over time, but I can do this change this weekend, thanks!

Comment (naomijub) [reply resources/entity.proto:28]: I dont think they will

File: src/gxtdb/adapters/entity.clj (hunk 2 above)
Comment (naomijub) [reply src/gxtdb/adapters/entity.clj:13]: I see, you are right https://docs.xtdb.com/clients/http/#_response_2

File: resources/entity.proto (hunk 1 above)
Comment (pfeodrippe) [reply resources/entity.proto:28]: Yeah, it seems very unlikely, will change them o/ Thanks!

Comment (pfeodrippe) [reply resources/entity.proto:20]: Done \o/

**Review by naomijub — APPROVED** [review]

**Pull request merged**